app/controlplane/internal/service/prometheus.go

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/biz"
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/jwt/apitoken"
+
+	jwtmiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
+	"github.com/prometheus/common/expfmt"
+)
+
+const (
+	// PrometheusMetricsPath is the path for the Prometheus metrics
+	PrometheusMetricsPath = "/prom/{org_name}/metrics"
+)
+
+// pattern is the regular expression pattern to match the organization name in the URL path
+var pattern = regexp.MustCompile(`/prom/([^/]+)/metrics`)
+
+// PrometheusService is the prometheus service
+type PrometheusService struct {
+	*service
+	// Use Cases
+	prometheusUseCase   *biz.PrometheusUseCase
+	organizationUseCase *biz.OrganizationUseCase
+}
+
+// NewPrometheusService creates a new prometheus service
+func NewPrometheusService(orgUseCase *biz.OrganizationUseCase, prometheusUseCase *biz.PrometheusUseCase, opts ...NewOpt) *PrometheusService {
+	return &PrometheusService{
+		organizationUseCase: orgUseCase,
+		prometheusUseCase:   prometheusUseCase,
+		service:             newService(opts...),
+	}
+}
+
+// MetricsRequestHandler is the handler for the metrics request. It fetches the Prometheus registry
+// and if found, retrieves all metrics in Prometheus format.
+func (p *PrometheusService) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Extract org_name from the URL path
+	match := pattern.FindStringSubmatch(r.URL.Path)
```

Comment:
I think that for this you can use morilla mux

for example

```
	orgName, ok := mux.Vars(r)["org_name"]
	if !ok {
		http.Error(w, "missing digest", http.StatusBadRequest)
		return
	}

```

Reply:
Uh nice, I'll change it, thanks!